lib/PuppeteerSharp/Browser.cs

```diff
@@ -182,6 +181,11 @@ public async ValueTask DisposeAsync()
             else
             {
                 await CloseAsync().ConfigureAwait(false);
+
+                if (Launcher.TempUserDataDir != null)
+                {
+                    await Launcher.TempUserDataDir.DisposeAsync().ConfigureAwait(false);
+                }
```

Comment:
It seems a bit odd to me why `Browser` should dispose a member of `Launcher` 🤔  
I experimented with this and how about we instead change `ExitedState` from

https://github.com/hardkoded/puppeteer-sharp/blob/eca16403b86f464f9182afae1b3bf2e4500950c1/lib/PuppeteerSharp/States/ExitedState.cs#L26-L27

to something like this?
```cs
if (p.TempUserDataDir is { } tempUserDataDir)
{
    tempUserDataDir
        .DisposeAsync()
        .AsTask()
        .ContinueWith(
            t => p.ExitCompletionSource.TrySetResult(true),
            TaskScheduler.Default);
}
else
{
    p.ExitCompletionSource.TrySetResult(true);
}
```

Reply:
I agree. I don't know why I accepted that State Machine refactor that now I have to maintain many years ago.

Reply:
Let me try to test that snippet. Thank you!